src/ApplicationInsights.Kubernetes/K8sInfoBootstrap.cs

```diff
@@ -51,10 +53,13 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                     await fetcher.UpdateK8sEnvironmentAsync(cancellationToken);
                 }
 
+                TimeSpan interval = _exponentialEmitter.GetNext();
                 _logger.LogDebug($"Finished update K8sEnvironment, next iteration will happen at {DateTime.UtcNow.Add(interval)} (UTC) by interval settings of {interval}");
 
-                // TODO: Consider using PeriodicTimer than delay below when move to .NET 6 +
-                await Task.Delay(interval, cancellationToken: cancellationToken);
+                using (PeriodicTimer timer = new PeriodicTimer(interval))
```

Comment:
If the timer is only used once, how is this any better than `Task.Delay()`?

Reply:
Actually a good point. I was thinking about having a periodic timer for a more accurate fixed interval. Now that it is exponential, there's no need for the timer.